overlord/devicestate/generate_rsa_key.go

```diff
@@ -0,0 +1,75 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package devicestate
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+var (
+	sshKeyFile       = "snapd.key.tmp"
+	sshPublicKeyFile = sshKeyFile + ".pub"
+)
```

Comment:
/run/snapd/ feels like the right place but I noticed that it isn't always created. For example, a test system of mine with no snaps installed does not have /run/snapd/ but it is created once a snap is installed.

I see it is referenced in dirs/dirs.go with the SnapRunNsDir variable. It is probably best that someone more active in snapd development advises you on if it is appropriate to use for your case.